File: lambda/hypothesis-generate/build_hypothesis_from_template.py
def build_hypothesis_from_template(input_data: dict) -> str:
    """Build If X / Then Y / Because Z hypothesis from structured form input."""
    audience = (input_data.get("targetAudience") or "").strip()
    audience_clause = f" for {audience}" if audience else ""

    direction = (input_data.get("direction") or "").strip().lower()
    expected_outcome = (input_data.get("expectedOutcome") or "").strip()
    y_clause = f"{expected_outcome} will {direction}"

    decision_metric = (input_data.get("decisionMetric") or "").strip()
    if decision_metric:
        y_clause += f", measured by {decision_metric}"

    proposed_change = (input_data.get("proposedChange") or "").strip()
    location = (input_data.get("location") or "").strip()
    reasoning = (input_data.get("reasoning") or "").strip()

    return (
        f"If {proposed_change} at {location}{audience_clause}, "
        f"then {y_clause}, because {reasoning}."
    )

File: lambda/hypothesis-generate/buildHypothesisFromTemplate.ts
export interface HypothesisInput {
  targetAudience: string
  location: string
  proposedChange: string
  userProblem: string
  expectedOutcome: string
  direction: string
  reasoning: string
  decisionMetric?: string
  guardrailMetric?: string
  notes?: string
}

export function buildHypothesisFromTemplate(input: HypothesisInput): string {
  const audienceClause = input.targetAudience.trim()
    ? ` for ${input.targetAudience.trim()}`
    : ''

  const direction = input.direction.trim().toLowerCase()
  let yClause = `${input.expectedOutcome.trim()} will ${direction}`

  if (input.decisionMetric?.trim()) {
    yClause += `, measured by ${input.decisionMetric.trim()}`
  }

  return `If ${input.proposedChange.trim()} at ${input.location.trim()}${audienceClause}, then ${yClause}, because ${input.reasoning.trim()}.`
}
